components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/TokenValidationHandler.java

```diff
@@ -638,11 +638,77 @@ private boolean hasRefreshTokenExpired(AccessTokenDO accessTokenDO) {
      */
     private AccessTokenDO findAccessToken(String tokenIdentifier) throws IdentityOAuth2Exception {
 
-        return OAuth2Util.getAccessTokenDOfromTokenIdentifier(tokenIdentifier);
+        String consumerKey = null;
+        OauthTokenIssuer oauthTokenIssuer = null;
+        if (isJWT(tokenIdentifier) || isIDTokenEncrypted(tokenIdentifier)) {
```

Comment:
Is this block necessary?
If we have a compressed JWT then isJWT() will throw an exception and break the flow